lib/makara/connection_wrapper.rb

```diff
@@ -144,38 +142,44 @@ def _makara_hijack
         def _makara_name
           #{@config[:name].inspect}
         end
-      }
+      RUBY
+
+      args = RUBY_VERSION >= "3.0.0" ? "..." : "*args, &block"
 
       # Each method the Makara::Proxy needs to hijack should be redefined in the underlying connection.
       # The new definition should allow for the proxy to intercept the invocation if required.
       @proxy.class.hijack_methods.each do |meth|
-        extension << %Q{
-          def #{meth}(*args, &block)
+        method_call = RUBY_VERSION >= "3.0.0" ? "public_send(#{meth.inspect}, ...)" : "#{meth}(*args, &block)"
```

Comment:
We need `public_send` here because `foo.bar=(...)` is not valid syntax in Ruby 3.0 (even though `foo.bar(...)` is)